src/main/java/com/velox/sloan/cmo/workflows/kapalibrary/SampleToPlateAutoAssigner.java

```diff
@@ -588,26 +609,38 @@ private void addPlateDimensionsToTaskOptions(int plateSize) throws RemoteExcepti
      * @throws com.velox.api.util.ServerException
      */
     private void assignPositionsMapFor96To384WellPlate(List<DataRecord> attachedSamples, String destination384PlateId, List<Map<String, Object>> quadrantValues, String targetPlateIdFieldName, String targetWellIdFieldName, List<DataRecord> newPlateProtocolRecordName) throws NotFound, RemoteException, com.velox.api.util.ServerException {
-        List<Map<String, Object>> newProtocolValues = new ArrayList<>();
         for (Map quadrantMap : quadrantValues) {
             String plateId = quadrantMap.get("PlateId").toString();
+            logInfo("quadrant PlateId: " + plateId);
             String quadrantId = quadrantMap.get("QuadrantNumber").toString();
             for (DataRecord sample : attachedSamples) {
                 String sampleId = sample.getStringVal("SampleId", user);
+                if (sample.getValue("RelatedRecord23", user) == null){
```

Comment:
if `RelatedRecord23 ` is null, that is plate id is invalid, the code will continue to execute? will cause NPE in the next `if` statement

Reply:
added a NotFound Exception to break the code if RelatedRecord23 is null.